app/src/main/java/org/mozilla/fenix/IntentReceiverActivity.kt

```diff
@@ -36,17 +35,10 @@ class IntentReceiverActivity : Activity() {
     }
 
     suspend fun processIntent(intent: Intent) {
-        if (!Browsers.all(this).isDefaultBrowser) {
-            /* If the user has unset us as the default browser, unset openLinksInAPrivateTab */
-            this.settings().openLinksInAPrivateTab = false
-            components.analytics.metrics.track(Event.PreferenceToggled(
-                preferenceKey = getString(R.string.pref_key_open_links_in_a_private_tab),
-                enabled = false,
-                context = applicationContext
-            ))
-        }
+        val settings = settings()
+        settings.unsetOpenLinksInAPrivateTabIfNecessary()
```

Comment:
The perf issue still affects app link, custom homescreen shortcuts, etc.: can you file a follow-up to see if we can improve this for these use cases?